### PR TITLE
Add initial checkstyle-checks project

### PR DESCRIPTION
**Pull request by Shredder121**
This includes the ArrayOrder check.

### CODE REVIEW

**Comment by Shredder121** [comment]
There are still some additions needed to the `pom.xml`.

**Comment by timowest** [comment]
Maybe also add the travis config

File: src/test/java/com/querydsl/checkstyle/checks/ArrayOrderTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        for (Class<?> clazz : classes) {
+            treeWalker.addChild(new DefaultConfiguration(clazz.getCanonicalName()));
+        }
+
+        checks.addChild(treeWalker);
+        return checks;
+    }
+
+    private static List<File> getFiles(Class<?>... classes) {
+        List<File> files = Lists.newArrayList();
```
Comment (timowest) [inline src/test/java/com/querydsl/checkstyle/checks/ArrayOrderTest.java]: Arrays.asList() instead?

**Comment by Shredder121** [comment]
> Maybe also add the travis config

Which Java platforms should I add? Oracle JDK 7 or also Oracle JDK 8?

**Comment by johnktims** [comment]
This looks good to me.

**Comment by Shredder121** [comment]
@timowest do you want to add the sonatype pom parent here or do you want this merged already?

**Comment by timowest** [comment]
The sonatype pom parent could be added.

**Comment by johnktims** [comment]
I've added the sonatype parent pom.

**Comment by Shredder121** [comment]
After the release of this project, we can merge the [checkstyle-arrayorder branch in the querydsl repository](https://github.com/querydsl/querydsl/compare/checkstyle-arrayorder).
Of course, then a different version than BUILD-SNAPSHOT will be needed.

**Comment by Shredder121** [comment]
So how does this look now?
I edited some formatting 'issues' that I saw looking back.
- Suffixed the checks with `Check`.
- More whitespace after the for loop's opening brace.

**Comment by timowest** [comment]
Looks ok to me.

**Pull request merged**